recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+
+
+class MongoCxxConan(ConanFile):
+    name = "mongo-cxx-driver"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://mongocxx.org"
+    description = "C++ Driver for MongoDB"
+    topics = ("conan", "libbsoncxx", "libmongocxx", "mongo", "mongodb", "database", "db")
+    settings = "os", "compiler", "arch", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = ("cmake", "cmake_find_package")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "polyfill": ["std", "boost", "mnmlstc", "experimental"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "polyfill": "boost"
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.options.polyfill == "mnmlstc":
+            # TODO: add mnmlstc support
+            # Cannot model mnmlstc (not packaged, is pulled dynamically) or
+            # std::experimental (how to check availability in stdlib?) polyfill
+            # dependencies
+            raise ConanInvalidConfiguration("mnmlstc is not yet supported")
+
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17" if self.options.polyfill == "std" else "11")
+
+    def requirements(self):
+        self.requires("mongo-c-driver/1.17.2")
+        if self.options.polyfill == "boost":
+            self.requires("boost/1.74.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-r" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BSONCXX_POLY_USE_MNMLSTC"] = self.options.polyfill == "mnmlstc"
+        self._cmake.definitions["BSONCXX_POLY_USE_STD_EXPERIMENTAL"] = self.options.polyfill == "experimental"
+        self._cmake.definitions["BSONCXX_POLY_USE_BOOST"] = self.options.polyfill == "boost"
+        self._cmake.definitions["BUILD_VERSION"] = self.version
+        self._cmake.definitions["BSONCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        self._cmake.definitions["MONGOCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        # FIXME: two CMake module/config files should be generated (mongoc-1.0-config.cmake and bson-1.0-config.cmake),
+        # but it can't be modeled right now.
+        # Fix should happen in mongo-c-driver recipe
+        shutil.copy("Findmongoc-1.0.cmake", "Findbson-1.0.cmake")
```

Comment:
As I said previously, I would recommend to check if `Findbson-1.0.cmake` doesn't already exist, because it might be fixed in `mongo-c-driver` in the future.

Reply:
Sorry, might have missed that. Fixed.